test_resolwe_runtime_utils.py

```diff
@@ -279,27 +294,154 @@ class TestCheckRC(ResolweRuntimeUtilsTestCase):
         self.assertJSONEqual(checkrc(1, ['Error']),
-                '{"proc.rc": 1, "proc.error": ["Error"]}')
+                             '{"proc.rc": 1, "proc.error": ["Error"]}')
 
     def test_improper_input(self):
-        self.assertEqual(checkrc(None), '{"proc.error": "Invalid return code: \'None\'."}')
-        self.assertEqual(checkrc('foo'), '{"proc.error": "Invalid return code: \'foo\'."}')
+        self.assertEqual(
+            checkrc(None), '{"proc.error": "Invalid return code: \'None\'."}')
+        self.assertEqual(
+            checkrc('foo'), '{"proc.error": "Invalid return code: \'foo\'."}')
         self.assertEqual(checkrc(1, 'foo', 'Error'),
                          '{"proc.error": "Invalid return code: \'foo\'."}')
         self.assertEqual(checkrc(1, None, 'Error'),
                          '{"proc.error": "Invalid return code: \'None\'."}')
 
+
+class ImportFileTestCase(TestCase):
+
+    _test_data_dir = os.path.join(TEST_ROOT, '.test_data')
+
+    def setUp(self):
+        # Clean after terminated tests
+        shutil.rmtree(self._test_data_dir, ignore_errors=True)
+
+        os.mkdir(self._test_data_dir)
+        os.chdir(self._test_data_dir)
+
+    def tearDown(self):
+        os.chdir(TEST_ROOT)
+        # shutil.rmtree(self._test_data_dir)
+
+    def _file(self, path):
+        """Add path prefix to test file."""
+        return os.path.join(TEST_ROOT, 'files', path)
+
+    def assertImportFile(self, src, dst, returned_name):
+        # Test both
+        src = self._file(src)
+        file_name = import_file(src, dst)
+        assert file_name == returned_name
+        assert os.path.exists(returned_name), "file not found"
+        assert os.path.exists(returned_name + '.gz'), "file not found"
+        os.remove(returned_name)
+        os.remove(returned_name + '.gz')
+
+        # Test extracted
+        file_name = import_file(src, dst, ImportedFormat.EXTRACTED)
+        assert file_name == returned_name
+        assert os.path.exists(returned_name), "file not found"
+        assert not os.path.exists(
+            returned_name + '.gz'), "file should not exist"
+        os.remove(returned_name)
+
+        # Test compressed
+        file_name = import_file(src, dst, ImportedFormat.COMPRESSED)
+        assert file_name == returned_name + '.gz'
+        assert os.path.exists(returned_name + '.gz'), "file not found"
+        assert not os.path.exists(returned_name), "file should not exist"
+        os.remove(returned_name + '.gz')
+
+    def test_uncompressed(self):
+        self.assertImportFile('some file.1.txt',
+                              'test uncompressed.txt',
+                              'test uncompressed.txt')
+
+    def test_gz(self):
+        self.assertImportFile('some file.1.txt.gz',
+                              'test gz.txt.gz',
+                              'test gz.txt')
+
+    def test_7z(self):
+        self.assertImportFile('some file.1.txt.zip',
```

Comment:
It seems that the input file is missing (see the error message on Travis). I get a similar error locally.

Reply:
Files added